### ISSUE TITLE
`sliceHack` not working in 2.7.0

### ISSUE THREAD

**Issue by Shohin93** [fabricjs/fabric.js#6127]
I upgraded fabric to `2.7.0` from `1.x`, and looks like `sliceHack` is not smoothing an image anymore, it gives a pixelated image. Did the syntax also change for that? 
```
 image.filters.push(
    new fabric.Image.filters.Resize({
      resizeType: 'sliceHack',
    })
  );
  // image.resizeFilter = new fabric.Image.filters.Resize({
  //   resizeType: 'sliceHack'
  // });
  image.set({
    scaleX: resizeScale,
    scaleY: resizeScale,
  });
  image.applyFilters();
```

**Comment by asturur** [comment]
As for any other issue or bug report, please add an active fiddle.
If you are updating now i suggest to jump on latest, and eventually wait for the version 4 if you are doing some with custom controls.

**Comment by sean-mystyle** [comment]
Fast forward to 2020... From my testing, SliceHack is working OK in 3.6.6 but not working in 4.2.0 - it works ok when the proportions are constrained, but from what I can tell but lockUniScaling = false does nothing and resize filters in general cause vertically squished images that are < 1 with scaleY to over-squish and display some near-half size scaleY instead of the correct size.  And, trying to disable vertical squish with lockUniScaling was not doing anything, so reverting back to 3.6.6 was the way to go as all work as expected in that version.  I will stick with 3.6.6 until resizeFilters and lockUniScaling are fixed in future 4.2.x versions...  Thank You

**Comment by asturur** [comment]
would you please open a fiddle from the standard template and open an issue with it so that i can track it? lockuniscaling i think has been addressed in a conversation, i should put it in the docs.
